File: common/htdocs/funcs/test/mmUserbase_test.php
<?php
   require_once("../mmConfig.inc");
   require_once("../mmUserbase.inc");
   $mmConfig = MMConfig::getInstance("../../../master_config.txt");
   $command_file = "mmUserbase_commands";
   if (! file_exists($command_file)) {
      echo "$command_file not found\n";
      exit;
   }
#   
#     Read all lines in a file into an array
#   
   $test_commands = file($command_file);
#   
   $userbase = new MM_Userbase();
   reset($test_commands);
   foreach ($test_commands as $cmd) {
      $cmdtrim = trim($cmd);
      if ($cmdtrim == "exit") {
         break;
      }
      $method_call = '$userbase->' . $cmdtrim;
      echo $method_call . "\n";
      eval ('$result = ' . $method_call . ';');
      if ($result === FALSE) {
         if ($userbase->exception_is_error()) {
            echo '   ERROR:   ' . $userbase->get_exception() . "\n";
         } else {
            echo '   INFO:    ' . $userbase->get_exception() . "\n";
         }
      } else {
         if (preg_match ('/^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\dZ$/',$result)) {
            $result = '__TIMESTAMP__';
         }
         echo "   OK   Result: ".$result."\n";
      }
#      echo "   STATUS: ".$userbase->debug_status()."\n";
   }
?>

File: common/lib/Metamod/test/mmUserbase_test.pl
#!/usr/bin/perl -w
use strict;
use File::Spec;
# small routine to get lib-directories relative to the installed file
use lib ('../../../lib', '../../../lib/Metamod');
use Metamod::Config;
use Metamod::mmUserbase;
my $mm_config  = Metamod::Config->new('../../../master_config.txt');
my $command_file = "../../../htdocs/funcs/test/mmUserbase_commands";
unless (-r $command_file) {die "Can not read from file: $command_file\n";}
open (COMMANDS,$command_file);
undef $/;
my $test_commands = <COMMANDS>;
my @test_commands = split(/\s*\n\s*/,$test_commands);
$/ = "\n"; 
close (COMMANDS);
my $userbase = Metamod::mmUserbase->new();
my $result;
foreach my $cmd (@test_commands) {
   if ($cmd eq "exit") {
      last;
   }
   my $method_call = '$userbase->' . $cmd;
   print $method_call . "\n";
   eval '$result = ' . $method_call . ';';
   if (!$result) {
      if ($userbase->exception_is_error()) {
         print '   ERROR:   ' . $userbase->get_exception() . "\n";
      } else {
         print '   INFO:    ' . $userbase->get_exception() . "\n";
      }
   } else {
      print "   OK   Result: ".$result."\n";
   }
}
